src/client/store/measures.js

```diff
@@ -1,5 +1,40 @@
 const measures = require('./measures.json')
 
 export const state = () => ({
-  measures: measures
+  measures: measures,
+  assignedMeasures: {}
 })
+
+export const mutations = {
+  assignMeasure(state, { id, measure }) {
+    const measures = state.assignedMeasures[id] || []
+
+    if(measures.includes(measure)) {
+      return
+    }
+
+    state.assignedMeasures = {
+      ...state.assignedMeasures,
+      [ id ]: [ ...measures, measure ]
+    }
+  },
+  unassignMeasure(state, { id, measure }) {
+    state.assignedMeasures = {
+      ...state.assignedMeasures,
+      [ id ]: [ ...state.assignedMeasures[id].filter(currentMeasure => currentMeasure.id !== measure.id) ]
```

Comment:
too much happening in this line. Move `state.assignedMeasures[id].filter(currentMeasure => currentMeasure.id !== measure.id` to a variable, then assign it here.